pkgs/by-name/ni/nixos-rebuild-ng/src/nixos_rebuild/nix.py

```diff
@@ -676,12 +676,37 @@ def switch_to_configuration(
         )
         cmd = []
 
-    run_wrapper(
-        [*cmd, path_to_config / "bin/switch-to-configuration", str(action)],
-        extra_env={"NIXOS_INSTALL_BOOTLOADER": "1" if install_bootloader else "0"},
-        remote=target_host,
-        sudo=sudo,
-    )
+    journalctl = None
+    unit = "nixos-rebuild-switch-to-configuration"
+    try:
+        journalctl = run_wrapper_bg(
+            [
+                which("journalctl"),
```

Comment:
Why do we need this `which` here? AFAIK `subprocess.run` already returns a good enough error when something is not found on PATH. But even if it didn't, I would prefer us to not change this in this PR and have a PR that replaces all `run_wrapper` usage with `which` otherwise we lose consistency here.